drivers/mlx90393/mlx90393.c

```diff
@@ -0,0 +1,625 @@
+    buffer[0] = MLX90393_COMMAND_WR;
+    buffer[1] = (uint8_t) (value >> 8);
+    buffer[2] = (uint8_t) (value & 0xFF);
+    buffer[3] = addr << 2;
+
+    int error = 0;
+
+    if ((error = _write_bytes(dev, buffer, 4)) != 0) {
+        return error;
+    }
+    if ((error = _check_status_byte(dev)) != 0) {
+        return error;
+    }
+    return 0;
+}
+
+static int _read_register(mlx90393_t *dev, uint8_t addr, uint16_t *value)
+{
+    uint8_t buffer_send[2];
+    buffer_send[0] = MLX90393_COMMAND_RR;
+    buffer_send[1] = addr << 2;
+    int error = 0;
+
+    if ((error = _write_bytes(dev, buffer_send, 2)) != 0) {
+        return error;
+    }
+    uint8_t buffer_receive[3];
+    if ((error = _read_bytes(dev, buffer_receive, 3)) != 0) {
+        return error;
+    }
+    if (buffer_receive[0] & MLX90393_STATUS_ERROR) {
+        return -EFAULT;
+    }
+    *value = (uint16_t)((buffer_receive[1] << 8) | buffer_receive[2]);
+
+    return 0;
+}
+
+static int _write_register_bits(mlx90393_t *dev, uint8_t addr, uint16_t mask, uint16_t value)
+{
+    uint16_t reg_value;
+    int error = 0;
+
+    if ((error = _read_register(dev, addr, &reg_value)) != 0) {
+        return error;
+    }
+    reg_value &= ~mask;
+    reg_value |= (mask & value);
+    if ((error = _write_register(dev, addr, reg_value)) != 0) {
+        return error;
+    }
+    return 0;
+}
+
+static int _calculate_temp(uint16_t raw_temp, uint16_t ref_temp)
+{
+    /* calculate temp in deci celsius (Application note MLX90393 temperature compensation - v4) */
+    return (MLX90393_TEMP_OFFSET + (((raw_temp - ref_temp) * 1000) / MLX90393_TEMP_RESOLUTION));
+}
+
+static int _get_gain_factor(mlx90393_gain_t gain)
+{
+    switch (gain)
+    {
+    case MLX90393_GAIN_5X:
+        return 500;
+    case MLX90393_GAIN_4X:
+        return 400;
+    case MLX90393_GAIN_3X:
+        return 300;
+    case MLX90393_GAIN_2_5X:
+        return 250;
+    case MLX90393_GAIN_2X:
+        return 200;
+    case MLX90393_GAIN_1_67X:
+        return 167;
+    case MLX90393_GAIN_1_33X:
+        return 133;
+    case MLX90393_GAIN_1X:
+        return 100;
+    default:
+        return -1;
+    }
+}
+
+static int _reset(mlx90393_t *dev)
+{
+    int error = 0;
+
+    if ((error = _write_byte(dev, MLX90393_COMMAND_RT)) != 0) {
+        return error;
+    }
+    if ((error = _check_status_byte(dev)) != 0) {
+        return error;
+    }
+    return 0;
+}
+
+static int _exit(mlx90393_t *dev)
+{
+    int error = 0;
+
+    if ((error = _write_byte(dev, MLX90393_COMMAND_EX)) != 0) {
+        return error;
+    }
+    if ((error = _check_status_byte(dev)) != 0) {
+        return error;
+    }
+    return 0;
+}
+
+static int _is_avaiable(mlx90393_t *dev)
+{
+    int error = 0;
+
+    if ((error = _write_register(dev, MLX90393_REG_CONN_TEST, CONN_TEST_DATA)) != 0) {
+        return error;
+    }
+    uint16_t buffer = 0x00;
+    if ((error = _read_register(dev, MLX90393_REG_CONN_TEST, &buffer)) != 0) {
```

Comment:
Variablendeklaration sollten für eine gute Lesbarkeit immer abgesetzt sein. Alternativ könnten Sie alle Variablendeklaration an den Anfang einer Funktion verschieben (klassischer ANSI-C Style).

```suggestion
    }

    uint16_t buffer = 0x00;

    if ((error = _read_register(dev, MLX90393_REG_CONN_TEST, &buffer)) != 0) {
```